### PR TITLE
Orca: Refactor Testing Structures (Ray-ctx)

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

Refactor orca testing structures. 

Several steps to go:
+ Refactor Pytest `Ray-ctx`
+ Refactor Pytest `Horovod`
    + Merge pytorch / tf tests?
    + Shorten test time
+ Refactor Pytest `Ray-Py37/Py38`
+ Remove unnecessary envs, maybe only need to keep 3
    + py37
    + py38
    + ctx (runner configuration is different)

Future Prospection:
+ Eliminate random fail
+ Minimize package requirements

This PR only does the first step.

### 2. Summary of the change 

Two parts

+ test scripts
+ location of testing python sciprts and configurations

### 4. How to test?

- [ ] Orca unit test

### CODE REVIEW

**Comment by hamham223** [comment]
Perhaps have a workflow try first @hkvision  :hugs:

File: python/orca/test/bigdl/orca/learn/ray/ctx/conftest.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/ray/ctx/conftest.py:1]: I don't think we need this file. There are already setup methods in each test file.

Comment (hamham223) [reply python/orca/test/bigdl/orca/learn/ray/ctx/conftest.py:1]: Still need `conf = {"spark.python.worker.reuse": "false"}`.

Comment (hamham223) [reply python/orca/test/bigdl/orca/learn/ray/ctx/conftest.py]: Removed, worth trying 😃

**Pull request merged**